### PR TITLE
Api refactoring with axios

### PR DESCRIPTION
**Pull request by fatb38**

### CODE REVIEW

**Review by TimCanu — COMMENTED** [review]
(review not finished at all, but I started this a few days ago so I prefer to post it now before forgetting)

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx [hunk 1]
```diff
@@ -100,15 +100,14 @@ function TeamEditDialog(props: TeamEditModalProps): JSX.Element {
...
-  const [teamName, setTeamName] = React.useState("");
-  const [teamPhone, setTeamPhone] = React.useState("");
-  const [teamEmail, setTeamEmail] = React.useState("");
-  const [addrLine1, setAddrLine1] = React.useState("");
-  const [addrLine2, setAddrLine2] = React.useState("");
-  const [addrZipCode, setAddrZipCode] = React.useState("");
-  const [addrCity, setAddrCity] = React.useState("");
-  const [addrCountry, setAddrCountry] = React.useState(auth.user?.settings?.country ?? "FR");
-
+  const [teamName, setTeamName] = React.useState(team?.name ?? "");
```
Comment (TimCanu) [inline packages/yourloops/pages/hcp/team-edit-dialog.tsx:103]: TODO: fix the issue of the dialog not being filled with the team info

File: packages/yourloops/lib/team/team-api.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      url: "/confirm/send/team/invite",
+      payload: { teamId, email, role },
+      config: { headers: { [HttpHeaderKeys.language]: getCurrentLang() } },
+    });
+    return data;
+  }
+
+  static async createTeam(team: Partial<ITeam>): Promise<ITeam> {
+    const { name, address, phone } = team;
+    if (!!name || !address || !!phone) {
```
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/lib/team/team-api.ts:112]: Yes, it was already present in the old API. We should move it later into the hook.

File: packages/yourloops/services/http.ts [hunk 3]
```diff
@@ -39,8 +39,13 @@ interface ArgsWithPayload<P> extends Args {
   payload?: P;
 }
 
+export enum StatusErrorMessage {
+  NotFound = "404-not-found"
+}
+
 export default class HttpService {
   private static retrieveAccessToken: () => Promise<string>;
+  static shorelineAccessToken: string;
```
Comment (TimCanu) [inline packages/yourloops/services/http.ts:48]: I suppose that we want to get rid of this but for now the API calls fail?

**Review by TimCanu — APPROVED** [review]

**Pull request merged**